.github/workflows/deploy-prod.yml

```diff
@@ -24,16 +39,77 @@ jobs:
           chmod 600 /home/runner/.ssh/google_compute_engine.pub
       - name: Authenticate with Google Cloud
         id: auth
-        uses: google-github-actions/auth@v1
+        uses: google-github-actions/auth@v2
         with:
           workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER }}
           service_account: ${{ secrets.GCP_SERVICE_ACCOUNT }}
           create_credentials_file: true
           cleanup_credentials: true
           export_environment_variables: true
       - name: Setup gcloud
-        uses: google-github-actions/setup-gcloud@v1
+        uses: google-github-actions/setup-gcloud@v2
+      - name: Setup Docker authentication
+        run: gcloud auth configure-docker ${{ env.REGION }}-docker.pkg.dev --quiet
+      - name: Get deployment configuration
+        run: |
+          gsutil cp gs://openreview-general/conf/deployment.conf ./deployment.conf
+          OPENREVIEW_TAG=$(grep 'OPENREVIEW_PY_V2_TAG=' deployment.conf | cut -d'"' -f2)
+          echo "OPENREVIEW_PY_VERSION=$OPENREVIEW_TAG" >> $GITHUB_ENV
+      - name: Set Environment Variables
+        run: |
+          echo "IMAGE_TAG_VERSION=${{ github.event.inputs.version }}" >> $GITHUB_ENV
+          echo "KFP_PIPELINE_NAME=${{ env.PIPELINE_NAME }}" >> $GITHUB_ENV
+          echo "IMAGE=${{ env.IMAGE_NAME }}" >> $GITHUB_ENV
+          echo "IMAGE_TAG=${{ env.REGION }}-docker.pkg.dev/${{ env.PROJECT }}/${{ env.REPO }}/${{ env.IMAGE_NAME }}" >> $GITHUB_ENV
+      - name: Check if image exists
+        id: image_check
+        run: |
+          IMAGE_PATH="${{ env.IMAGE_TAG }}:${{ env.IMAGE_TAG_VERSION }}"
+          echo "Checking for image: ${IMAGE_PATH}"
+          
+          if docker manifest inspect "${IMAGE_PATH}" > /dev/null 2>&1; then
+            echo "exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "exists=false" >> $GITHUB_OUTPUT
+          fi
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
```

Comment:
Does it matter what version of python we use? I think we used 3.11 in other parts of the PR.

Reply:
I don't think it matters since we're only building the pipeline, but I agree to use 3.11 for consistency